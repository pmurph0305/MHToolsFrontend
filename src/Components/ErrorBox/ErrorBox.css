.ErrorBox {
    background-color: lightcoral;
    display: inline-block;
    width: fit-content;
    margin: 1rem;
    padding: 0.5rem;
}