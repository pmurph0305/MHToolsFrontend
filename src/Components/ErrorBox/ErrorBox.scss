@import "../../styles/shared.scss";

.ErrorBox {
  background-color: $error-background-color;
  display: block;
  width: fit-content;
  margin: auto;
  padding: 0.5rem;
}
