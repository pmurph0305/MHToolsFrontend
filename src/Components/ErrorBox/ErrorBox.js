import React from 'react';

import './ErrorBox.css'

const ErrorBox = ({error}) => {
    var displayedError;
    if (error && error.length) {
        // Removes error passed from database.
        displayedError = error.split('error:')
        // Still console log the actual database error in console though.
        if (displayedError[1]) {
            console.log(displayedError[1]);
        }
        displayedError = displayedError[0].toString();
    } else if (error) {
        // otherwise it's a type error of some kind.
        displayedError = error.toString();
    }

    return (
        <div>
            {displayedError
            ? <div class='ErrorBox'>
                {displayedError}
              </div>
            : null
            }
        </div>
    )
}

export default ErrorBox