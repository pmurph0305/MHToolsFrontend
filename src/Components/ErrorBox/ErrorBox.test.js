import { shallow } from 'enzyme';
import React from 'react';
import ErrorBox from './ErrorBox'


describe('ErrorBox tests', () => {
    let wrapper;
    beforeEach(() => {
        let mockError = 'Error'
        wrapper = shallow(<ErrorBox error={mockError}/>)
    })
    
    it('Exists', () => {
        expect(wrapper.exists()).toBe(true);
    })

    it('Matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('Displays error', () => {
        expect(wrapper.text()).toEqual('Error');
    })

    it('Splits error display', () => {
        // Error that error box splits.
        const mockErrorSplit = 'This is a test error error: Actual Error'
        // wrapper to test.
        const wrapperSplit = shallow(<ErrorBox error={mockErrorSplit}/>)
        // make sure displayed error is correctly split.
        expect(wrapperSplit.text()).toEqual("This is a test error");
    })

    it('Handles null and undefined error', () => {
        // Mocks for null and undefined.
        let mockNull = null;
        let mockUndefined = undefined;
        // wrapper for null error
        const wrapNull = shallow(<ErrorBox error={mockNull}/>)
        // make sure the box still exists but has no error displayed
        expect(wrapNull.exists()).toBe(true);
        expect(wrapNull.text()).toEqual("");
        // wrapper for undefined error.
        const wrapUndefined = shallow(<ErrorBox error={mockUndefined}/>)
         // make sure the box still exists but has no error displayed
        expect(wrapUndefined.exists()).toBe(true);
        expect(wrapUndefined.text()).toEqual("");
    })
})