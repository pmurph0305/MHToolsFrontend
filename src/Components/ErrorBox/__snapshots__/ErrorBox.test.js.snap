// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox tests Matches snapshot 1`] = `ShallowWrapper {}`;
